Code/Tools/AssetProcessor/native/AssetManager/assetProcessorManager.cpp

```diff
@@ -3782,33 +3802,114 @@ namespace AssetProcessor
                 QStringList resolvedDependencyList;
                 QString resolvedDatabaseName;
 
-                if (!ResolveSourceFileDependencyPath(jobDependency.m_jobDependency.m_sourceFile, resolvedDatabaseName, resolvedDependencyList))
+                if (!ResolveSourceFileDependencyPath(
+                        jobDependency.m_jobDependency.m_sourceFile, resolvedDatabaseName, resolvedDependencyList))
                 {
                     continue;
                 }
 
                 for (const auto& thisEntry : resolvedDependencyList)
                 {
                     SourceFileDependencyEntry newDependencyEntry(
-                        builderId,
+                        builderId, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(), thisEntry.toUtf8().constData(),
+                        SourceFileDependencyEntry::DEP_JobToJob, // significant line in this code block
+                        false);
+                    newDependencies.push_back(AZStd::move(newDependencyEntry));
+                }
+
+                // Source dependencies don't have any concept of jobs so if we store an entry for every job, we end up with duplicates.
+                // This isn't an issue with the builder, so no error/warning is needed, just check to avoid duplicates.
+                if (auto result = jobDependenciesDeduplication.emplace(builderId, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(),
+                    resolvedDatabaseName.toUtf8().constData()); result.second)
+                {
+                    SourceFileDependencyEntry newDependencyEntry(
+                        builderId, entry.m_sourceFileInfo.m_databasePath.toUtf8().constData(), resolvedDatabaseName.toUtf8().constData(),
+                        jobDependency.m_jobDependency.m_sourceFile.m_sourceDependencyType ==
```

Comment:
Maybe pull this out to a separate line stored in a local variable, instead of needing to comment in-line that it's significant?